service/src/main/java/bio/terra/externalcreds/services/JwtUtils.java

```diff
@@ -0,0 +1,147 @@
+package bio.terra.externalcreds.services;
+
+import bio.terra.externalcreds.config.ExternalCredsConfig;
+import bio.terra.externalcreds.models.GA4GHPassport;
+import bio.terra.externalcreds.models.GA4GHVisa;
+import bio.terra.externalcreds.models.GA4GHVisa.Builder;
+import bio.terra.externalcreds.models.LinkedAccount;
+import bio.terra.externalcreds.models.LinkedAccountWithPassportAndVisas;
+import bio.terra.externalcreds.models.TokenTypeEnum;
+import com.google.common.annotations.VisibleForTesting;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.shaded.json.JSONObject;
+import com.nimbusds.jwt.JWTParser;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.sql.Timestamp;
+import java.text.ParseException;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtDecoders;
+import org.springframework.security.oauth2.jwt.JwtException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class JwtUtils {
+  public static final String PASSPORT_JWT_V11_CLAIM = "passport_jwt_v11";
+  public static final String GA4GH_PASSPORT_V1_CLAIM = "ga4gh_passport_v1";
+  public static final String GA4GH_VISA_V1_CLAIM = "ga4gh_visa_v1";
+  public static final String VISA_TYPE_CLAIM = "type";
+  public static final String JKU_HEADER = "jku";
+
+  private final ExternalCredsConfig externalCredsConfig;
+
+  public JwtUtils(ExternalCredsConfig externalCredsConfig) {
+    this.externalCredsConfig = externalCredsConfig;
+  }
+
+  public LinkedAccountWithPassportAndVisas enrichAccountWithPassportAndVisas(
+      LinkedAccount linkedAccount, OAuth2User userInfo) {
+    var passportJwtString = userInfo.<String>getAttribute(PASSPORT_JWT_V11_CLAIM);
+    if (passportJwtString != null) {
+      var passportJwt = decodeJwt(passportJwtString);
+
+      var visaJwtStrings =
+          Objects.requireNonNullElse(
+              passportJwt.getClaimAsStringList(GA4GH_PASSPORT_V1_CLAIM),
+              Collections.<String>emptyList());
+
+      var visas =
+          visaJwtStrings.stream().map(s -> buildVisa(decodeJwt(s))).collect(Collectors.toList());
```

Comment:
When I can I like to chain these calls, e.g. 
```suggestion
          visaJwtStrings.stream().map(this::decodeJwt).map(this::buildVisa).collect(Collectors.toList());
```